californium-core/src/test/java/org/eclipse/californium/core/test/MemoryLeakingHashMapTest.java

```diff
@@ -364,11 +364,21 @@ public CoapObserverAndCanceler(final CountDownLatch latch, final AtomicBoolean e
 			this.cancelProactively = cancelProactively;
 		}
 
-		public void setObserveRelation(CoapObserveRelation relation) {
+		public synchronized void setObserveRelation(CoapObserveRelation relation) {
 			this.relation = relation;
 		}
 
 		public void onLoad(CoapResponse response) {
+			CoapObserveRelation relation;
+			synchronized (this) {
```

Comment:
The latch is always initialized with a given number (of notifications to expect) in the corresponding test cases.
I guess the tests work properly as long as the time to wait for the latch to count down is sufficiently long, which sems to be the case with the *generously* added 500 ms and a notification interval of 50 ms ....